ai_summary/indico_ai_summary/client/components/ind_summarize_button.jsx

```diff
@@ -0,0 +1,186 @@
+// them and/or modify them under the terms of the MIT License;
+// see the LICENSE file for more details.
+
+// Main entry point of the summarization feature in the Indico plugin.
+// This file defines the Summarize button, manages modal states,
+// handles prompt selection, generates summaries, and integrates
+// with Indico's API to fetch and save meeting notes.
+import React, {useState, useEffect} from 'react';
+import ReactDOM from 'react-dom';
+import {Modal, Button, Form, Grid, GridRow, GridColumn, Header, Card, CardContent} from 'semantic-ui-react';
+import {Translate} from 'indico/react/i18n';
+import {getSavedPrompts, deletePrompt, getSummary, saveSummary, clearSummary} from '../utils/storage';
+import {fetchSummary, fetchEventNote, saveSummaryToEvent} from '../services/summarize';
+import {buildPrompt} from '../utils/prompts';
+import PromptSelector from './prompt_selector';
+import SummaryPreview from './summary_preview';
+import ManagePromptsModal from './manage_prompts_modal';
+import '../styles/ind_summarize_button.module.scss';
+
+function SummarizeButton({eventId}) {
+  // React State Definitions
+  const [open, setOpen] = useState(false); // modal visibility
+  const [selectedPromptKey, setSelectedPromptKey] = useState('default'); // selected prompt type
+  const [customPromptText, setCustomPromptText] = useState(''); // custom prompt content
+  const [editedPromptText, setEditedPromptText] = useState(''); // manual edits to generated prompt
+  const [savedPrompts, setSavedPrompts] = useState([]); // saved custom prompts
+  const [summaryHtml, setSummaryHtml] = useState(''); // generated summary HTML
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(null);
+  const [saving, setSaving] = useState(false); // saving summary indicator
+  const [manageModalOpen, setManageModalOpen] = useState(false); // saved prompts modal
+
+  useEffect(() => { // load saved prompts and last generated summary from localStorage
+    setSavedPrompts(getSavedPrompts());
+    setSummaryHtml(getSummary());
+  }, []);
+
+  // trigger summary generation by calling the backend API
+  const runSummarize = async () => {
+    try {
+      setLoading(true);
+      setError(null);
+      setSummaryHtml('');
+
+      // build prompt: either edited, saved, or from template
+      const prompt =
+      editedPromptText ||
+      (selectedPromptKey.startsWith('saved-')
+        ? savedPrompts[parseInt(selectedPromptKey.replace('saved-', ''), 10)]
+        : buildPrompt(selectedPromptKey, customPromptText));
+      
+      // call backend to fetch summary
+      const data = await fetchSummary(eventId, prompt);
+      // if response contains the summary - display and cache it
+      if (data.summary_html) {
+        setSummaryHtml(data.summary_html);
+        saveSummary(data.summary_html);
+      } else {
+        setError('No summary returned.');
+      }
+    } catch (e) {
```

Comment:
It's a good practise to only put code that is expected to throw into a try block (`fetchSummary` in this case). Otherwise it can hide other issues with the code.